### PR TITLE
Es2 refactoring

### PR DESCRIPTION
**Pull request by joaojunior**
This Pull Request is only the code in the Pull Request https://github.com/django-haystack/django-haystack/pull/1336. But, here I refactoring the code and resolve the conflicts.
I send the message to @PedroAquilino, that made the PR #1336 to ask about the conflicts and to ask about to use your code in this PR and he say ok.

---

This change is [Reviewable](https://reviewable.io/reviews/django-haystack/django-haystack/1391)

### CODE REVIEW

File: test_haystack/solr_tests/server/get-solr-download-url.py [hunk 1]
```diff
@@ -27,9 +27,11 @@
...
+    download_url_mirror = urljoin(mirror_data['preferred'], mirror_data['path_info'])
     # The Apache mirror script's response format has recently changed to exclude the actual file paths:
-    if not download_url.endswith(tarball):
-        download_url = urljoin(download_url, dist_path)
-
+    if not download_url_mirror.endswith(tarball):
+        download_url_mirror = urljoin(download_url_mirror, dist_path)
+    response = requests.head(download_url_mirror)
+    if response.status_code == 200:
+        download_ulr = download_url_mirror
```
Comment (acdha) [inline test_haystack/solr_tests/server/get-solr-download-url.py]: That's a typo for `download_ulr`

**Comment by acdha** [comment]
Review status: 0 of 11 files reviewed at latest revision, 5 unresolved discussions, some commit checks failed.

---

*[.travis.yml, line 30 \[r2\]](https://reviewable.io:443/reviews/django-haystack/django-haystack/1391#-KLbYKxT5yKb-avgBGnq:-KLbYKxT5yKb-avgBGnr:2130896771) ([raw file](https://github.com/django-haystack/django-haystack/blob/efab1a5d02b34d1d44d116582ec1445ab2921900/.travis.yml#L30)):*
> ```YAML
>     - mkdir -p $HOME/download-cache
>     - >
>       if [[ $VERSION_ES == '>=2.0.0,<3.0.0' ]];
> ```

Has the Travis CI team white-listed the ES 2 repos yet? We can install 1.7 from APT and I'd generally prefer we do that, although this isn't very much code

---

*[.travis.yml, line 42 \[r2\]](https://reviewable.io:443/reviews/django-haystack/django-haystack/1391#-KLbXyW1ks0e2tugeHfz:-KLbXyW2494KbmQ5DcMX:-216318683) ([raw file](https://github.com/django-haystack/django-haystack/blob/efab1a5d02b34d1d44d116582ec1445ab2921900/.travis.yml#L42)):*
> ```YAML
> 
> install:
>     - if [[ $TRAVIS_PYTHON_VERSION == '2.6' ]]; then pip install mock==1.0.1; fi
> ```

We no longer support Python 2.6 so this can be removed

---

*[test_haystack/settings.py, line 98 \[r2\]](https://reviewable.io:443/reviews/django-haystack/django-haystack/1391#-KLbYjChyCLw0eb_Vp3M:-KLbYjChyCLw0eb_Vp3N:872317455) ([raw file](https://github.com/django-haystack/django-haystack/blob/efab1a5d02b34d1d44d116582ec1445ab2921900/test_haystack/settings.py#L98)):*
> ```Python
> 
> if os.getenv('VERSION_ES') == ">=2.0.0,<3.0.0":
>     HAYSTACK_CONNECTIONS['elasticsearch'] = {
> ```

If we're not customizing the other parameters, perhaps we should make that more obvious by setting the `ENGINE` key directly?

---

*[test_haystack/elasticsearch_tests/__init__.py, line 15 \[r2\]](https://reviewable.io:443/reviews/django-haystack/django-haystack/1391#-KLbZ2GtBwETSX0GVUzd:-KLbZ2GtBwETSX0GVUze:-192506418) ([raw file](https://github.com/django-haystack/django-haystack/blob/efab1a5d02b34d1d44d116582ec1445ab2921900/test_haystack/elasticsearch_tests/__init__.py#L15)):*
> ```Python
>         import elasticsearch
>         if not ((1, 0, 0) <= elasticsearch.__version__ < (2, 0, 0)):
>             raise ImportError
> ```

What do you think about at least adding logging here to make it more obvious that we're skipping the tests due to an unrecognized version of elasticsearch-py? The regular skip message below seems likely to confuse someone in the future.

---

*[test_haystack/solr_tests/server/get-solr-download-url.py, line 36 \[r1\]](https://reviewable.io:443/reviews/django-haystack/django-haystack/1391#-KLbXOQeA1ojzywxDb_u-r1-36:-KLbZdkVUCvudZr1QACT:-698197794) ([raw file](https://github.com/django-haystack/django-haystack/blob/5c24066d9ee6fd589078af950ea60bd6a0e7ef81/test_haystack/solr_tests/server/get-solr-download-url.py#L36)):*
<details><summary><i>Previously, acdha (Chris Adams) wrote…</i></summary>
> `download_ulr` is a typo
</details>
This change should no longer be necessary with https://github.com/django-haystack/django-haystack/commit/89ef54a556e48f4597292eaf23f942284208785a

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/django-haystack/django-haystack/1391)*

**Comment by acdha** [comment]
Reviewed 5 of 11 files at r1.
Review status: 5 of 11 files reviewed at latest revision, 6 unresolved discussions, some commit checks failed.

---

*[test_haystack/elasticsearch2_tests/__init__.py, line 15 \[r2\]](https://reviewable.io:443/reviews/django-haystack/django-haystack/1391#-KLb_VcETjnEWW2r_gKz:-KLb_VcETjnEWW2r_gL-:-789627638) ([raw file](https://github.com/django-haystack/django-haystack/blob/efab1a5d02b34d1d44d116582ec1445ab2921900/test_haystack/elasticsearch2_tests/__init__.py#L15)):*
> ```Python
>         import elasticsearch
>         if not ((2, 0, 0) <= elasticsearch.__version__ < (3, 0, 0)):
>             raise ImportError
> ```

Same comment as below about wanting to make it clear why the installed `elasticsearch` module is not being used

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/django-haystack/django-haystack/1391)*

**Comment by joaojunior** [comment]
Review status: 1 of 11 files reviewed at latest revision, 6 unresolved discussions.

---

*[.travis.yml, line 30 \[r2\]](https://reviewable.io:443/reviews/django-haystack/django-haystack/1391#-KLbYKxT5yKb-avgBGnq:-KMakvGXgR7koUsl8t-C:-499146009) ([raw file](https://github.com/django-haystack/django-haystack/blob/efab1a5d02b34d1d44d116582ec1445ab2921900/.travis.yml#L30)):*
<details><summary><i>Previously, acdha (Chris Adams) wrote…</i></summary>
> Has the Travis CI team white-listed the ES 2 repos yet? We can install 1.7 from APT and I'd generally prefer we do that, although this isn't very much code
</details>
Done.

---

*[.travis.yml, line 42 \[r2\]](https://reviewable.io:443/reviews/django-haystack/django-haystack/1391#-KLbXyW1ks0e2tugeHfz:-KMakwMIBoT-6uIA_Cx5:-499146009) ([raw file](https://github.com/django-haystack/django-haystack/blob/efab1a5d02b34d1d44d116582ec1445ab2921900/.travis.yml#L42)):*
<details><summary><i>Previously, acdha (Chris Adams) wrote…</i></summary>
> We no longer support Python 2.6 so this can be removed
</details>
Done.

---

*[test_haystack/settings.py, line 98 \[r2\]](https://reviewable.io:443/reviews/django-haystack/django-haystack/1391#-KLbYjChyCLw0eb_Vp3M:-KMakxIKd1cy7tFGRE7j:-499146009) ([raw file](https://github.com/django-haystack/django-haystack/blob/efab1a5d02b34d1d44d116582ec1445ab2921900/test_haystack/settings.py#L98)):*
<details><summary><i>Previously, acdha (Chris Adams) wrote…</i></summary>
> If we're not customizing the other parameters, perhaps we should make that more obvious by setting the `ENGINE` key directly?
</details>
Done.

---

*[test_haystack/elasticsearch2_tests/__init__.py, line 15 \[r2\]](https://reviewable.io:443/reviews/django-haystack/django-haystack/1391#-KLb_VcETjnEWW2r_gKz:-KMakyGrnjjStfC0Aat6:-499146009) ([raw file](https://github.com/django-haystack/django-haystack/blob/efab1a5d02b34d1d44d116582ec1445ab2921900/test_haystack/elasticsearch2_tests/__init__.py#L15)):*
<details><summary><i>Previously, acdha (Chris Adams) wrote…</i></summary>
> Same comment as below about wanting to make it clear why the installed `elasticsearch` module is not being used
</details>
Done.

---

*[test_haystack/elasticsearch_tests/__init__.py, line 15 \[r2\]](https://reviewable.io:443/reviews/django-haystack/django-haystack/1391#-KLbZ2GtBwETSX0GVUzd:-KMaktqFK7wkkRg0hF9g:-499146009) ([raw file](https://github.com/django-haystack/django-haystack/blob/efab1a5d02b34d1d44d116582ec1445ab2921900/test_haystack/elasticsearch_tests/__init__.py#L15)):*
<details><summary><i>Previously, acdha (Chris Adams) wrote…</i></summary>
> What do you think about at least adding logging here to make it more obvious that we're skipping the tests due to an unrecognized version of elasticsearch-py? The regular skip message below seems likely to confuse someone in the future.
</details>
Done.

---

*[test_haystack/solr_tests/server/get-solr-download-url.py, line 36 \[r1\]](https://reviewable.io:443/reviews/django-haystack/django-haystack/1391#-KLbXOQeA1ojzywxDb_u-r1-36:-KMaPEYZXcyotbFU4NR8:-499146009) ([raw file](https://github.com/django-haystack/django-haystack/blob/5c24066d9ee6fd589078af950ea60bd6a0e7ef81/test_haystack/solr_tests/server/get-solr-download-url.py#L36)):*
<details><summary><i>Previously, acdha (Chris Adams) wrote…</i></summary>
> This change should no longer be necessary with https://github.com/django-haystack/django-haystack/commit/89ef54a556e48f4597292eaf23f942284208785a
</details>
Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/django-haystack/django-haystack/1391)*

**Comment by joaojunior** [comment]
@acdha Why this PR don't run travis-ci now?

**Comment by acdha** [comment]
That is odd – https://travis-ci.org/django-haystack/django-haystack/requests says “.travis.yml is missing and builds without .travis.yml are disabled” but https://github.com/joaojunior/django-haystack/blob/es2_refactoring/.travis.yml is obviously there.

**Comment by acdha** [comment]
Ah, it's a misleading message from Travis CI – according to https://lint.travis-ci.org/ there's a syntax error.

File: .travis.yml [hunk 2]
```diff
@@ -51,6 +65,7 @@ matrix:
 
 services:
     - elasticsearch
+        - python: "pypy"
```
Comment (acdha) [inline .travis.yml]: This is apparently what's causing the Travis build failures

**Comment by pior** [comment]
@acdha, what's the status on this PR?
What's missing / what must be fixed?

**Comment by acdha** [comment]
@pior Right now, more testing & review. If you have time to work on this, please update this issue with your results.

**Comment by seocam** [comment]
This has been replaced by #1460 and can be closed.

**Comment by acdha** [comment]
I just uploaded v2.6.0 to PyPI:

https://django-haystack.readthedocs.io/en/v2.6.0/changelog.html#v2-6-0-2017-01-02 
https://pypi.python.org/pypi/django-haystack/2.6.0

**Pull request closed without merging**